### ISSUE TITLE
Use the `cryptography` crate to simplify coordinated upgrades?

### ISSUE THREAD

**Issue by tarcieri** [RustCrypto/meta#5]
In 3rd party projects where I'm consuming RustCrypto crates, it's pretty tricky to determine when an upgrade can be performed, even for myself.

This PR is a good example: there are new versions of `crypto-mac` and `hmac` out, but `pbkdf2` wasn't updated yet:
https://github.com/RustCrypto/password-hashes/pull/44

For users dealing with these inconsistencies, I think it'd be nice if there were a single "compatibility number" they could focus on, and I think the `cryptography` crate's version would be a good one.

Concretely I'd propose to start doing this in RustCrypto projects that are both the most high-level/user-facing and which require coordinated upgrades of multiple trait crates and/or combine multiple algorithm crates, and in particular crates where user-facing trait-based composition is desired.

I think such crates within RustCrypto could depend on the `cryptography` crate, source the traits they use from it, and publish a "This crate is compatible with `cryptography` v0.3" message in the README.md and `rustdoc`.

This may sound bad because we'd need to bump the `cryptography` version as a dependency every time we cut a release, but I argue that's a feature, not a bug: it means uncoordinated, focused upgrade work can occur in part of the project and end users can easily see that they can't yet perform an upgrade because all of their dependent crates haven't been bumped to the new `cryptography` version.

tl;dr: by coordinating upgrades through the `cryptography` crate, end users have only a single version number to focus on in order to determine compatibility

**Comment by newpavlov** [comment]
Personally, I am not a fan of this approach. It could be a stop-gap solution until 1.0 releases, but I don't think it should be a permanent one.

I think we are better to find a way to get notifications about crates which use outdated dependencies. For example, by setting-up a meta-crate which would include all crates in this organization and use `cargo outdated` inside a periodic CI job. Previously we had deps.rs badges, but unfortunately the website does not work anymore.

**Comment by tarcieri** [comment]
> It could be a stop-gap solution until 1.0 releases, but I don't think it should be a permanent one.

After 1.0 this problem largely goes away

> I think we are better to find a way to get notifications about crates which use outdated dependencies. 

I really like Dependabot for this, especially as it provides a lot of context for dependency changes like auto-linking to changelogs

**Comment by newpavlov** [comment]
>I really like Dependabot for this

Ah, yes, I forgot about it. We probably should setup it for all repositories.

**Comment by tarcieri** [comment]
Yeah, sounds good. All that needs to be done is copying https://github.com/RustCrypto/traits/blob/master/.github/dependabot.yml into the other repos (possibly customizing it a bit first before we propagate it everywhere)

**Comment by tarcieri** [comment]
This has been implemented (and we migrated to the `crypto` crate)
